src/main/java/com/limechain/storage/trie/TrieStorage.java

```diff
@@ -96,12 +99,13 @@ private NodeData getNodeFromDb(byte[] nodeMerkleValue, Nibbles key) {
         return getNodeFromDb(childMerkleValue, remainder.drop(1));
     }
 
-    private TrieNodeData getTrieNodeFromMerkleValue(@NotNull byte[] childMerkleValue) {
+    @Nullable
+    public TrieNodeData getTrieNodeFromMerkleValue(@NotNull byte[] childMerkleValue) {
```

Comment:
why is this public?

Reply:
The method is crucial for the logic of the `DiskTrieService`. One of the performance drivers is that after traversing the cache layer we don't traverse the disk up until the last ancestor, but we directly jump to it via this method and continue traversal from there.